lib/stripe/webhook.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Stripe.Webhook do
+  @moduledoc """
+  Creates a Stripe Event from webhook's payload if signature is valid.
+
+  Use `construct_event/3` to verify the authenticity of a webhook request and
+  convert its payload into a `Stripe.Event` struct.
+
+    case Stripe.Webhook.construct_event(payload, signature, secret) do
+      {:ok, %Stripe.Event{} = event} ->
+        # Return 200 to Stripe and handle event
+      {:error, reason} ->
+        # Reject webhook by responding with non-2XX
+    end
+  """
+  @default_tolerance 300
+  @expected_scheme "v1"
+
+  def construct_event(payload, signature_header, secret, tolerance \\ @default_tolerance) do
+    case verify_header(payload, signature_header, secret, tolerance) do
+      :ok ->
+        {:ok, Poison.decode!(payload)}
```

Comment:
The documentation here specifies the function will return an `{:ok, %Stripe.Event{}}`, but it's actually returning an `{:ok, %{}}`. The map keys are atoms, but it is not an event struct.

We would have to return `{:ok, Stripe.Converter.convert_result(payload)}` for the output to match the documentation.

Is there a reason you went with `Poison.decode!/1` instead? I ask because it's certainly possible our `Stripe.Event` struct does not map something it should. If that's the case, in the interest of merging this, I'm ok with you updating the documentation instead, so it specifies it's returning a map, and then submitting an issue if the struct has a problem, which we can work on.